src/Subscriber/Response/PaymentMethodRouteResponseSubscriber.php

```diff
@@ -0,0 +1,182 @@
+<?php declare(strict_types=1);
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <shopware@adyen.com>
+ */
+
+namespace Adyen\Shopware\Subscriber\Response;
+
+use Adyen\Shopware\Provider\AdyenPluginProvider;
+use Adyen\Shopware\Service\PaymentMethodsFilterService;
+use Adyen\Shopware\Service\PaymentMethodsService;
+use Adyen\Shopware\Struct\AdyenPaymentMethodDataStruct;
+use Shopware\Core\Checkout\Payment\PaymentMethodEntity;
+use Shopware\Core\Checkout\Payment\SalesChannel\PaymentMethodRouteResponse;
+use Shopware\Core\PlatformRequest;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Contracts\Service\ResetInterface;
+
+class PaymentMethodRouteResponseSubscriber implements EventSubscriberInterface, ResetInterface
+{
+    /**
+     * @var AdyenPluginProvider
+     */
+    private $adyenPluginProvider;
+
+    /**
+     * @var PaymentMethodsService
+     */
+    private $paymentMethodsService;
+
+    /**
+     * @var PaymentMethodsFilterService
+     */
+    private $paymentMethodsFilterService;
+
+    /**
+     * @var array
+     */
+    private $paymentMethodsResponse;
+
+    public function __construct(
+        AdyenPluginProvider $adyenPluginProvider,
+        PaymentMethodsService $paymentMethodsService,
+        PaymentMethodsFilterService $paymentMethodsFilterService
+    ) {
+        $this->adyenPluginProvider = $adyenPluginProvider;
+        $this->paymentMethodsService = $paymentMethodsService;
+        $this->paymentMethodsFilterService = $paymentMethodsFilterService;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            // Ensure event is executed before "Shopware\Core\System\SalesChannel\Api\StoreApiResponseListener".
+            KernelEvents::RESPONSE => ['adjustPaymentMethodsResponse', 11000],
```

Comment:
@peterojo I have for now done it with a response subscriber as I think this data is only relevant for that one Store API response route? 

If you want to have the same data on other responses that might include the payment method etc. I would need to use a different event or a service decoration. Both are possible but I think this should be fine?